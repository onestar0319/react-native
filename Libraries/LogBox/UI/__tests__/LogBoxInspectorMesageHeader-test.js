/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow strict-local
 */

'use strict';

const React = require('react');
const LogBoxInspectorMessageHeader = require('../LogBoxInspectorMessageHeader')
  .default;
const render = require('../../../../jest/renderer');

describe('LogBoxInspectorMessageHeader', () => {
  it('should render error', () => {
    const output = render.shallowRender(
      <LogBoxInspectorMessageHeader
        title="Error"
        level="error"
        collapsed={false}
        message={{
          content: 'Some error message',
          substitutions: [],
        }}
        onPress={() => {}}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render fatal', () => {
    const output = render.shallowRender(
      <LogBoxInspectorMessageHeader
        title="Fatal Error"
        level="fatal"
        collapsed={false}
        message={{
          content: 'Some fatal message',
          substitutions: [],
        }}
        onPress={() => {}}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render syntax error', () => {
    const output = render.shallowRender(
      <LogBoxInspectorMessageHeader
        title="Syntax Error"
        level="syntax"
        collapsed={false}
        message={{
          content: 'Some syntax error message',
          substitutions: [],
        }}
        onPress={() => {}}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should not render See More button for short content', () => {
    const output = render.shallowRender(
      <LogBoxInspectorMessageHeader
        title="Warning"
        level="warn"
        collapsed={false}
        message={{
          content: 'Short',
          substitutions: [],
        }}
        onPress={() => {}}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should not render "See More" if expanded', () => {
    const output = render.shallowRender(
      <LogBoxInspectorMessageHeader
        title="Warning"
        level="warn"
        collapsed={false}
        message={{content: '#'.repeat(301), substitutions: []}}
        onPress={() => {}}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render "See More" if collapsed', () => {
    const output = render.shallowRender(
      <LogBoxInspectorMessageHeader
        title="Warning"
        level="warn"
        collapsed={true}
        message={{
          content: '#'.repeat(301),
          substitutions: [],
        }}
        onPress={() => {}}
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
