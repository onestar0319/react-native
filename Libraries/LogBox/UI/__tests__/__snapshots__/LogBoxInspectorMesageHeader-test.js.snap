// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxInspectorMessageHeader should not render collapse button for short content 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "elevation": 2,
      "flex": 0,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 5,
        "marginTop": 10,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 20,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 28,
          },
          Object {
            "color": "rgba(250, 186, 48, 1)",
          },
        ]
      }
    >
      Warning
    </Text>
  </View>
  <Text
    numberOfLines={null}
    style={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontSize": 14,
        "fontWeight": "500",
        "includeFontPadding": false,
        "lineHeight": 20,
        "paddingBottom": 10,
        "paddingHorizontal": 12,
      }
    }
  >
    <LogBoxMessage
      message={
        Object {
          "content": "Short",
          "substitutions": Array [],
        }
      }
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.6)",
        }
      }
    />
  </Text>
</View>
`;

exports[`LogBoxInspectorMessageHeader should not render error 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "elevation": 2,
      "flex": 0,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 5,
        "marginTop": 10,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 20,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 28,
          },
          Object {
            "color": "rgba(243, 83, 105, 1)",
          },
        ]
      }
    >
      Error
    </Text>
  </View>
  <Text
    numberOfLines={null}
    style={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontSize": 14,
        "fontWeight": "500",
        "includeFontPadding": false,
        "lineHeight": 20,
        "paddingBottom": 10,
        "paddingHorizontal": 12,
      }
    }
  >
    <LogBoxMessage
      message={
        Object {
          "content": "Short",
          "substitutions": Array [],
        }
      }
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.6)",
        }
      }
    />
  </Text>
</View>
`;

exports[`LogBoxInspectorMessageHeader should render "collapse" if expanded 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "elevation": 2,
      "flex": 0,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 5,
        "marginTop": 10,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 20,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 28,
          },
          Object {
            "color": "rgba(250, 186, 48, 1)",
          },
        ]
      }
    >
      Warning
    </Text>
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(69, 69, 69, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 3,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 12,
          }
        }
      >
        collapse
      </Text>
    </LogBoxButton>
  </View>
  <Text
    numberOfLines={null}
    style={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontSize": 14,
        "fontWeight": "500",
        "includeFontPadding": false,
        "lineHeight": 20,
        "paddingBottom": 10,
        "paddingHorizontal": 12,
      }
    }
  >
    <LogBoxMessage
      message={
        Object {
          "content": "########################################################################################################################################################################################################",
          "substitutions": Array [],
        }
      }
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.6)",
        }
      }
    />
  </Text>
</View>
`;

exports[`LogBoxInspectorMessageHeader should render "see more" if collapsed 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "elevation": 2,
      "flex": 0,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 5,
        "marginTop": 10,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 20,
            "fontWeight": "600",
            "includeFontPadding": false,
            "lineHeight": 28,
          },
          Object {
            "color": "rgba(250, 186, 48, 1)",
          },
        ]
      }
    >
      Warning
    </Text>
    <LogBoxButton
      backgroundColor={
        Object {
          "default": "transparent",
          "pressed": "rgba(69, 69, 69, 1)",
        }
      }
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 3,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 12,
            "fontWeight": "300",
            "lineHeight": 12,
          }
        }
      >
        see more
      </Text>
    </LogBoxButton>
  </View>
  <Text
    numberOfLines={5}
    style={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontSize": 14,
        "fontWeight": "500",
        "includeFontPadding": false,
        "lineHeight": 20,
        "paddingBottom": 10,
        "paddingHorizontal": 12,
      }
    }
  >
    <LogBoxMessage
      message={
        Object {
          "content": "########################################################################################################################################################################################################",
          "substitutions": Array [],
        }
      }
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.6)",
        }
      }
    />
  </Text>
</View>
`;
